src/librustc/hir/map/def_collector.rs

```diff
@@ -283,4 +284,17 @@ impl<'a> visit::Visitor<'a> for DefCollector<'a> {
             _ => visit::walk_stmt(self, stmt),
         }
     }
+
+    fn visit_token(&mut self, t: Token) {
+        if let Token::Interpolated(nt) = t {
+            match nt.0 {
+                token::NtExpr(ref expr) => {
+                    if let ExprKind::Mac(..) = expr.node {
```

Comment:
I believe this should just be `token::Expr(ref expr) => self.visit_expr(expr),`.

Ideally, we'd add `visit_interpolated` and `walk_interpolated` (a la https://github.com/sinkuu/rust/blob/c0ccab4c2330659b67edb5889d7f9ee702aebbe3/src/libsyntax/fold.rs#L618) -- could you add a FIXME for this?

Reply:
It should, but doesn't work due to lifetimes:(

Reply:
I believe you can fix the lifetime error by chaging `fn visit_token` to take the token by reference instead of by value.

Reply:
Wait, nvm -- you can get a reference to the token with an appropriate lifetime from a token stream, so that won't work. I think what you have is fine for now.

Reply:
I cound't find out how to get `&'a Token` from `&'a TokenStream`, and `&'a TokenStream` from `&'a ThinTokenStream`, since all of their methods return owned type by cloning underlying `Rc`. I tried simply adding `_ref` variant of their methods but failed, because it's impossible to construct `&TokenStreamKind::JointTree(TokenTree)` from `&TokenTree` for example.